src/WebJobs.Extensions.NotificationHubs/NotificationHubAttribute.cs

```diff
@@ -39,12 +39,14 @@ public sealed class NotificationHubAttribute : Attribute
         /// Optional. A string value indicating the app setting to use as the Notification Hubs connection
         /// string, if different than the one specified in the <see cref="NotificationHubsConfiguration"/>.
         /// </summary>
+        [AutoResolve(AllowTokens = false)]
         public string ConnectionStringSetting { get; set; }
 
         /// <summary>
         /// Optional. The Notification Hub Name to use, if different than the one specified in the
         /// <see cref="NotificationHubsConfiguration"/>.
         /// </summary>
+        [AutoResolve(AllowTokens = false)]
```

Comment:
So using AllowTokens=false seems wrong -- that would require it to be a lookup value. If you want to support both literal strings as well as replacements, we should just have this be [AutoResolve] and when the picker works is done, it needs to write %appsetting%.

---
In reply to: [70492852](https://github.com/Azure/azure-webjobs-sdk-extensions/pull/120#discussion_r70492852) [](ancestors = 70492852)